conn/physic/units_test.go

```diff
@@ -536,3 +538,257 @@ func BenchmarkCelsiusFloatg(b *testing.B) {
 		buf.Reset()
 	}
 }
+
+func TestAtod(t *testing.T) {
+	const (
+		negative = true
+		positive = false
+	)
+	succeeds := []struct {
+		in       string
+		expected decimal
+		n        int
+	}{
+		{"123456789", decimal{"123456789", 0, positive}, 9},
+		{"1nM", decimal{"1", 0, positive}, 1},
+		{"2.2nM", decimal{"22", -1, positive}, 3},
+		{"12.5mA", decimal{"125", -1, positive}, 4},
+		{"-12.5mA", decimal{"125", -1, negative}, 5},
+		{"1ma1", decimal{"1", 0, positive}, 1},
+		{"+1ma1", decimal{"1", 0, positive}, 2},
+		{"-0.00001%rH", decimal{"1", -5, negative}, 8},
+		{"0.00001%rH", decimal{"1", -5, positive}, 7},
+		{"1.0", decimal{"1", 0, positive}, 3},
+		{"0.10001", decimal{"10001", -5, positive}, 7},
+		{"-0.10001", decimal{"10001", -5, negative}, 8},
+		{"1n", decimal{"1", 0, positive}, 1},
+		{"200n", decimal{"2", 2, positive}, 3},
+		{".01", decimal{"1", -2, positive}, 3},
```

Comment:
For completeness, can you add "+.01" and "-.01". It's really more as a documentation than anything else.

Reply:
Np. Found another bug introduced by last fix.